Enable negative cache when using `--cache` (#757)
awslabs/mountpoint-s3: doc/SEMANTICS.md

When this option is provided, Mountpoint will check S3 to ensure the object exists and return the latest object content.
Unlike other file systems, Mountpoint does not support setting the `O_DIRECT` flag via `fcntl` after the file has been opened.

> When caching is enabled, Mountpoint can also remember queries for missing keys. Once you try to
> access a file that does not exist on S3, subsequent attempts (within the configured TTL) may still
> not detect it, even if it was independently added to S3.

Reviewer: ```suggestion
When caching is enabled, Mountpoint also remembers when objects do *not* exist. Once you try to
access a file that does not exist on S3, subsequent attempts (within the configured TTL) may still
fail, even if it was later added to S3.
```
